File: Leetcode Problems/Two Sum/two_sum.py
from typing import List

class Solution:
    def two_sum(self, nums:List[int], target: int) -> List[int]:
        num_indicies = {}
        for i, num in enumerate(nums):
            difference = target - num
            if difference in num_indicies:
                return[num_indicies[difference], i]
            num_indicies[num] = i

if __name__ == "__main__":
    solution = Solution()
    print(solution.two_sum([2, 7, 11, 15], 9))

File: go/twoSum.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {

	hashMap := make(map[int]int)

	for i, num := range nums {

		diff := target - num

		if index, exists := hashMap[diff]; exists {
			return []int{index, i}
		}

		hashMap[num] = i
	}

	return []int{}
}

func main() {
	nums := []int{2, 7, 11, 15}
	target := 9

	fmt.Println(twoSum(nums, target))
}